### ISSUE TITLE
Call for the implement code of BYOL, MoCo and Swav 

### ISSUE THREAD

**Issue by Trent-tangtao** [facebookresearch/dino#34]
Thanks for your excellent work! 
And in your paper,  you made a lot of comparisons with BYOL, Moco and Swav,
I am grateful if you can release your implement code of BYOL, moco and swav.

**Comment by mathildecaron31** [comment]
Hi @Trent-tangtao 

Thanks for your kind words. In order to keep this codebase simple, we have not been planning to release my implementation for BYOL, MoCo and SwAV. The repo is meant to be a simple implementation for DINO and related transfer tasks only and not a generic SSL library.

In addition there are already official releases for these works so I am not sure of the added value in releasing the re-implementation. I encourage you to take a look at https://github.com/facebookresearch/vissl, https://github.com/facebookresearch/swav or https://github.com/facebookresearch/moco.

I apologize for the possible inconvenience, and feel free to reach out if you have any questions.

**Comment by mathildecaron31** [comment]
Hi @KyleZheng1997 

1. I used temperature = 0.1
2. I've experimented with different parameters and the same schedule as used in this repo was the optimal (0.996 to 1 with cosine)
3. 65k, though it was giving only minor improvement over 4k for example
4. learning rate is the same as in this repo. For weight decay keeping it fixed to 0.05 was working the best in my experiments.
5. no clipping
6. 3 layers and final dimension at 256. hidden unit at 2048
7. I use synchronous BN in projection head
8. batch size is 1024 just like in DINO experiments
9. same as in this repo
10. symmetric version

Hope that helps :)

**Comment by Hiusam** [comment]
Hi, thanks for your excellent work! I have the following questions:
1. Do you use AdamW for MoCoV2 training? And what are the parameters of the optimizer?
2. What activation function did you use in the MLP head?
3. Do you use the activation function after the final layer of the MLP head?

**Comment by mathildecaron31** [comment]
1- AdamW (exactly like DINO training)
2- GeLU (exactly like DINO training)
3- No activation after final layer. The features are l2 normalized (like in MoCo paper)
4- 72.7% top-1 with linear eval after 800 epochs (71.6% after 300 epochs) (please see Tables 2 and 13 from our DINO paper). I have not reported top5.

Hope that helps!

**Comment by mathildecaron31** [comment]
In case that helps here are the checkpoints at 300 epochs used in our paper:
BYOL:
https://dl.fbaipublicfiles.com/dino/byol_vitsmall16_300ep_pretrain.pth

SwAV:
https://dl.fbaipublicfiles.com/dino/swav_vitsmall16_300ep_pretrain.pth

MoCo-v2:
https://dl.fbaipublicfiles.com/dino/moco_vitsmall16_300ep_pretrain.pth

DINO:
https://dl.fbaipublicfiles.com/dino/dino_vitsmall16_300ep_pretrain.pth
